Framework/Algorithms/src/SofQWCentre.cpp

```diff
@@ -111,24 +113,12 @@ void SofQWCentre::exec() {
   m_EmodeProperties.initCachedValues(*inputWorkspace, this);
   int emode = m_EmodeProperties.m_emode;
 
-  // Get a pointer to the instrument contained in the workspace
-  Instrument_const_sptr instrument = inputWorkspace->getInstrument();
-
-  // Get the distance between the source and the sample (assume in metres)
-  IComponent_const_sptr source = instrument->getSource();
-  IComponent_const_sptr sample = instrument->getSample();
-  V3D beamDir = sample->getPos() - source->getPos();
+  const auto &detectorInfo = inputWorkspace->detectorInfo();
+  const auto &spectrumInfo = inputWorkspace->spectrumInfo();
+  V3D beamDir = detectorInfo.samplePosition() - detectorInfo.sourcePosition();
```

Comment:
I am not sure about that yet. It has little to do with detectors. Maybe making it a method of the (no yet existing) new instrument would make more sense. The same goes for L1, which we *did* add, but that is used very frequently so it was useful. We should however also consider whether or not it should stay, or be moved to another class.